### PR TITLE
Fixes #13041: rudder-metrics-reporting fail to report reports on directive/expected reports in 4.3, as the database schema changed

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/13041

### CODE REVIEW

File: rudder-webapp/SOURCES/rudder-metrics-reporting [hunk 1]
```diff
@@ -143,6 +146,14 @@ def metrics():
   cmd="[ -x /usr/bin/dpkg ] && dpkg-query -W rudder-server-root || rpm -q rudder-server-root"
   data["package_version"] = command(["/bin/sh", "-c", cmd]).strip()
 
+  # Report mode
+  cmd="grep 'RUDDER_REPORT_MODE' /var/rudder/cfengine-community/inputs/rudder.json | sed 's/.*"RUDDER_REPORT_MODE":"\(.*\)",.*/\1/'"
```
Comment (peckpeck) [inline rudder-webapp/SOURCES/rudder-metrics-reporting]: this regex work with current json, but not necessarily if formatting change
I would remove the coma and add optional spaces around the semicolon

Comment (amousset) [reply rudder-webapp/SOURCES/rudder-metrics-reporting]: should we load rudder.json as a python object instead?

Comment (ncharles) [reply rudder-webapp/SOURCES/rudder-metrics-reporting]: this is exactly the regexp used in rudder agent info. 

@amousset do you thinkg we would really benefit from reading the json ?

Comment (ncharles) [reply rudder-webapp/SOURCES/rudder-metrics-reporting]: ok

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**